### PR TITLE
adds message param to ban, deban, and deflag worker methods to make them work properly

### PR DESCRIPTION
**Pull request by BeccaSheldon**
@arvel @mrgordon I added a message to these methods so that they work as expected.

### CODE REVIEW

**Comment by mrgordon** [comment]
It looks like the message should be called `reason` for ban and deflag: https://github.com/dolores/builder/blob/develop/app/controllers/workers.rb#L246

Interestingly deban doesn't seem to take a message at all: https://github.com/dolores/builder/blob/develop/app/controllers/workers.rb#L254

**Comment by BeccaSheldon** [comment]
@arvel @syrnick @mrgordon if flag and deflag look good, I'll merge those with develop and checkout a new branch to remove the ban and deban admin methods.

**Comment by arvel** [comment]
+1 for the flag methods, we can probably just remove the ban methods

**Comment by smathieu** [comment]
Should persist be an option on `deflag` as well?

**Comment by mrgordon** [comment]
No `deflag` automatically removes the persisted flag for the user/worker pair. I can't imagine you would want to leave it persisted, but feel free to let us know if you see a use for that.

File: lib/crowdflower/worker.rb [hunk 1]
```diff
@@ -44,12 +44,17 @@ def notify(worker_id, message)
...
 
-    def flag( worker_id, reason = nil )
-      connection.put( "#{resource_uri}/#{worker_id}/flag", :body => { :reason => reason }, :headers => { "Content-Length" => "0" } )
+    def flag( worker_id, reason = nil, persist = false )
+      params = {
+      :headers => { "Content-Length" => "0" }, 
+      :reason => reason, 
+      :persist => persist
+      }
+      connection.put( "#{resource_uri}/#{worker_id}/flag", :query => params )
```
Comment (smathieu) [inline lib/crowdflower/worker.rb]: Shouldn't this be `:body => params`?

Comment (smathieu) [inline lib/crowdflower/worker.rb]: In fact, this should be:

``` ruby
    def flag( worker_id, reason = nil, persist = false )
      params = {
        :reason => reason,
        :persist => persist
      }
      connection.put( "#{resource_uri}/#{worker_id}/flag", :body => params, :headers => { "Content-Length" => "0" } )
    end
```

Comment (mrgordon) [inline lib/crowdflower/worker.rb]: Agreed

**Pull request merged**